004-quarkus-HHH-and-HV/src/main/java/io/quarkus/qe/hibernate/items/ItemsResource.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.qe.hibernate.items;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Root;
+import javax.transaction.Transactional;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+@Path("/items")
+@Transactional
+public class ItemsResource {
+
+    @Inject
+    EntityManager em;
+
+    @GET
+    @Path("/count")
+    public int countOrders() {
+        CriteriaBuilder cb = em.getCriteriaBuilder();
+        CriteriaQuery<Item> cq = cb.createQuery(Item.class);
+        Root<Item> root = cq.from(Item.class);
+        Join<Item, Customer> custJoin = (Join<Item, Customer>) root.fetch(Item_.customer);
```

Comment:
```suggestion
        Fetch<Item, Customer> custFetch = root.fetch("customer");
        Join<Item, Customer> custJoin = (Join<Item, Customer>) custFetch;;
```

Didn't work with Criteria API before, but my point of ^^^ is to get rid of the red syntax highlighting.

Also, it looks like Join is triggered eagerly, but actually never used then, so I would try to change `countOrders` to something like `itemsList` and utilize that JOIN as a Predicate for CriteriaQuery. Because now method name `countOrders` is misleading and the actual reproducer happens while JOIN

Correct me if I'm wrong :dancers:

Reply:
With that suggestion applied there is no need for <artifactId>hibernate-jpamodelgen</artifactId>

Reply:
I've already changed as you suggested. But I kept the count orders as it's easier to assert in the test. If you strongly think that we shoult return a list, feel free to re request it as another change and I will do it.

Reply:
I think it will be more clear what is the actual reproducer is. Plus this is a public repo and maybe someone will find inspiration by that ;D

Reply:
I've made some changes to make it clear what is intended here and also added a comment to note that this is necessary.
Let me know if it looks better to you.